src/puppetlabs/puppetdb/http/factsets.clj

```diff
@@ -1,16 +1,31 @@
 (ns puppetlabs.puppetdb.http.factsets
   (:require [net.cgrand.moustache :refer [app]]
+            [puppetlabs.puppetdb.http :as http]
             [puppetlabs.puppetdb.http.facts :as facts]
             [puppetlabs.puppetdb.http.query :as http-q]
             [puppetlabs.puppetdb.middleware :refer [verify-accepts-json validate-query-params
                                                     wrap-with-paging-options wrap-with-parent-check]]
             [puppetlabs.puppetdb.query.paging :as paging]
-            [puppetlabs.puppetdb.query-eng :refer [produce-streaming-body]]))
+            [puppetlabs.puppetdb.query-eng :as eng]))
+
+(defn factset-status
```

Comment:
So more or less, I'm sticking with the same pattern for catalogs & reports ... however ... there is a problem.

Actually, queries could be considered valid for singletons, that is - some queries do make sense, like extraction queries that limit what parameters are returned. If we agree this to be true, then all these endpoints need fixing.